### PR TITLE
Update loss ops for conditional models

### PR DESCRIPTION
**Pull request by Aniket1998**
Each generator and discriminator now has a property loss_type that determines whether it requires labels from the training set, randomly generated labels, or no labels at all. The loss ops now accept the labels alongwith the inputs from the trainer and proceeds according to the loss_type of the generator and discriminator

### CODE REVIEW

File: torchgan/losses/boundaryequilibrium.py [hunk 1]
```diff
@@ -102,16 +102,30 @@ def update_k(self, loss_real, loss_fake):
         elif self.k > 1.0:
             self.k = 1.0
 
-    def train_ops(self, generator, discriminator, optimizer_discriminator, real_inputs, device, labels_provided=False):
+    def train_ops(self, generator, discriminator, optimizer_discriminator,
+            real_inputs, batch_size, device, labels=None):
         if self.override_train_ops is not None:
-            return self.override_train_ops(generator, discriminator, optimizer_discriminator,
-                                           real_inputs, device, labels_provided)
+            return self.override_train_ops(self, generator, discriminator, optimizer_discriminator,
```
Comment (avik-pal) [inline torchgan/losses/boundaryequilibrium.py]:
```suggestion
            return self.override_train_ops(generator, discriminator, optimizer_discriminator,
```

File: torchgan/losses/boundaryequilibrium.py [hunk 2]
```diff
@@ -102,16 +102,30 @@ def update_k(self, loss_real, loss_fake):
...
+                   real_inputs, labels)
```
Comment (avik-pal) [inline torchgan/losses/boundaryequilibrium.py]:
```suggestion
                   real_inputs, labels, device)
```

File: torchgan/losses/loss.py [hunk 3]
```diff
@@ -8,7 +8,7 @@ class GeneratorLoss(nn.Module):
 
     Args:
         reduction (string, optional): Specifies the reduction to apply to the output.
-            If `none` no reduction will be applied. If `elementwise_mean` the sum of
+        If `none` no reduction will be applied. If `elementwise_mean` the sum of
```
Comment (avik-pal) [inline torchgan/losses/loss.py]:
```suggestion
            If `none` no reduction will be applied. If `elementwise_mean` the sum of
```

File: torchgan/losses/loss.py [hunk 4]
```diff
@@ -43,17 +59,30 @@ def __init__(self, reduction='elementwise_mean', override_train_ops=None):
...
         self.override_train_ops = override_train_ops
 
-    def train_ops(self, generator, discriminator, optimizer_discriminator, real_inputs, device,
-                  labels_provided=False):
+    def train_ops(self, generator, discriminator, optimizer_discriminator, real_inputs, batch_size, device,
+                  labels=None):
         if self.override_train_ops is not None:
-            return self.override_train_ops(generator, discriminator, optimizer_discriminator,
-                   real_inputs, labels_provided)
+            return self.override_train_ops(self, generator, discriminator, optimizer_discriminator,
```
Comment (avik-pal) [inline torchgan/losses/loss.py]:
```suggestion
            return self.override_train_ops(generator, discriminator, optimizer_discriminator,
```

File: torchgan/losses/loss.py [hunk 5]
```diff
@@ -43,17 +59,30 @@ def __init__(self, reduction='elementwise_mean', override_train_ops=None):
...
+                   real_inputs, labels)
```
Comment (avik-pal) [inline torchgan/losses/loss.py]:
```suggestion
                   real_inputs, labels, device)
```

File: torchgan/losses/wasserstein.py [hunk 6]
```diff
@@ -158,17 +158,26 @@ def forward(self, interpolate, d_interpolate):
         # relying on autograd
         return wasserstein_gradient_penalty(interpolate, d_interpolate, self.reduction)
 
-    def train_ops(self, generator, discriminator, optimizer_discriminator, real_inputs, device, labels_provided=False):
+    def train_ops(self, generator, discriminator, optimizer_discriminator,
+            real_inputs, batch_size, device, labels=None):
         if self.override_train_ops is not None:
-            return self.override_train_ops(generator, discriminator, optimizer_discriminator, real_inputs, device,
-                                           labels_provided)
+            return self.override_train_ops(self, generator, discriminator, optimizer_discriminator,
```
Comment (avik-pal) [inline torchgan/losses/wasserstein.py]:
```suggestion
            return self.override_train_ops(generator, discriminator, optimizer_discriminator,
```

File: torchgan/losses/wasserstein.py [hunk 7]
```diff
@@ -158,17 +158,26 @@ def forward(self, interpolate, d_interpolate):
...
+                   real_inputs, labels)
```
Comment (avik-pal) [inline torchgan/losses/wasserstein.py]:
```suggestion
                   real_inputs, labels, device)
```
